### PR TITLE
Support Ed255192020 keys, open badges example

### PR DESCRIPTION
**Pull request by cykoder**

### CODE REVIEW

File: src/utils/vc/crypto/Ed25519VerificationKey2020.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      return new this(b58.decode(verificationMethod.publicKeyBase58));
+    }
+
+    if (verificationMethod.publicKeyBase64) {
+      return new this(base64.decode(verificationMethod.publicKeyBase64));
+    }
+
+    throw new Error(`Unsupported signature encoding for ${Ed255192020VerKeyName}`);
+  }
+}
```
Comment (lovesh) [inline src/utils/vc/crypto/Ed25519VerificationKey2020.js:29]: `verifier` function needs to be overwritten as well.

File: tests/integration/issuing.test.js [hunk 2]
```diff
@@ -122,13 +139,20 @@ describe('Verifiable Credential issuance where issuer has a Dock DID', () => {
...
     expect(result.verified).toBeTruthy();
   }, 40000);
 
+  test('(JWT) Issue a verifiable credential with ed25519 key and verify it (2020)', async () => {
+    const issuerKey = getKeyDoc(issuer1DID, dock.keyring.addFromUri(issuer1KeySeed, null, 'ed25519'), 'Ed25519VerificationKey2020');
+    const credential = await issueCredential(issuerKey, unsignedCred, true, null, null, null, null, false, 'jwt');
+    const result = await verifyCredential(credential, { resolver });
+    expect(result.verified).toBeTruthy();
+  }, 40000);
+
```
Comment (lovesh) [inline tests/integration/issuing.test.js:155]: Can you add a test for presentation as well with this new scheme?

**Review by mike-parkhill — DISMISSED** [review]

File: src/utils/vc/custom_crypto.js [hunk 3]
```diff
@@ -35,12 +38,15 @@ export {
...
   Ed25519SigName,
+  Ed255192020VerKeyName,
+  Ed255192020SigName,
   Sr25519VerKeyName,
   Sr25519SigName,
   EcdsaSecp256k1VerificationKey2019,
   EcdsaSecp256k1Signature2019,
   Ed25519VerificationKey2018,
   Ed25519Signature2018,
+  Ed25519Signature2020,
```
Comment (lovesh) [inline src/utils/vc/custom_crypto.js:49]: `/contexts/security_context` needs update?

**Review by lovesh — CHANGES_REQUESTED** [review]

File: src/utils/vc/crypto/Ed25519VerificationKey2020.js (hunk 1 above)
Comment (cykoder) [reply src/utils/vc/crypto/Ed25519VerificationKey2020.js:29]: it doesn't seem to need to be in order to work, it behaves same as ed25519 2018

File: tests/integration/issuing.test.js (hunk 2 above)
Comment (cykoder) [reply tests/integration/issuing.test.js:155]: done

File: src/utils/vc/custom_crypto.js (hunk 3 above)
Comment (cykoder) [reply src/utils/vc/custom_crypto.js:49]: no, this key has its own context. wont work outside of our SDK if we put 2020 context in there

File: src/utils/vc/crypto/Ed25519VerificationKey2020.js (hunk 1 above)
Comment (lovesh) [reply src/utils/vc/crypto/Ed25519VerificationKey2020.js:29]: Do you expect people reading this code to know this? Why not keep it clear?

Comment (cykoder) [reply src/utils/vc/crypto/Ed25519VerificationKey2020.js:29]: it seemed pretty obvious given it extends Ed25519VerificationKey2018 while all others dont typically extend from another defined class

Comment (cykoder) [reply src/utils/vc/crypto/Ed25519VerificationKey2020.js:29]: have updated with a comment at the end of the class

**Review by lovesh — APPROVED** [review]

**Pull request merged**